src/openscad.cc

```diff
@@ -311,13 +311,25 @@ int cmdline(const char *deps_output_file, const std::string &filename, const std
 
 	// Do some minimal checking of output directory before rendering (issue #432)
 	fs::path output_directory = fs::path(output_file_str).parent_path();
+	if (output_directory.generic_string() == "") {
+		output_directory = fs::current_path();
+	}
 	if (!fs::is_directory(output_directory)) {
 		PRINTB(
 			"%s is not a directory (output file: %s). - Skipping\n",
 			output_directory.c_str() % output_file_str.c_str()
 		);
 		return 1;
 	}
+	// Create a file in the directory to confirm that its at least writable (issue #432)
+	fs::path temp_file = output_directory / fs::path("_tmp_open_scad.tmp");
+	std::ofstream temp_fstream(temp_file.c_str());
```

Comment:
It looks like the windows mxe builds don't like this.  I think you need to call `.generic_string()` first.  That will return a `std::string` which should also be fine to send to ofstream constructor without `.c_str()`

Also, couldn't we just test opening `new_output_file` directly instead of writing and deleting a temp file?

Reply:
And if ~~`new_output_file`~~ `filename_str` is opened here in append mode, then it should even help keep previous file contents, if any, just in case the render process fails somehow before it starts writing the new results.

Reply:
I will switch it over to .generic_string().  Perhaps what I did was only valid in c++ 2017.  I didn't want to touch the actual filename because as you suggest, it might already exist.  I *think* that opening it in append would be fine but I didn't want to touch/contaminate the existing file.  Are you ok leaving it as is?